fuel/src/test/kotlin/com/github/kittinunf/fuel/RequestHeaderTest.kt

```diff
@@ -46,4 +47,22 @@ class RequestHeaderTest : BaseTestCase() {
         assertThat(string, containsString(headerValue))
     }
 
+    @Test
+    fun multipleHeadersByTheSameKeyWillBeCorrectlyFormatted(){
+        val request = manager.request(Method.GET, "/get")
+                .header("foo" to "bar","a" to "b", "cookie" to "val1=x", "cookie" to "val2=y","cookie" to "val3=z", "cookie" to "val4=j")
+
+        assertEquals("[ val1=x,val2=y,val3=z,val4=j ]",request.headers["cookie"] )
```

Comment:
I think these brackets are just here for the `toString` and not actually when you send this request?, if so, that should be made clear in a comment or something like that :)

Reply:
@SleeplessByte  thanks for the feed back I can remove the brackets and the code that adds them.

 to be clear these are NOT how the request should be formatted?

`cookie: [foo=bar,buzz=fizz]`
 
should be 

`cookie: foo=bar,buzz=fizz`

I can update this pr would you comment on other changes?

Reply:
Hi @markGilchrist , No, you got it! I commented on the complete PR. Rest looked great.

I would add a space. It's not wrong to not add a space, but the format allows for it, so it's more readable: i.e. this would be `Cookie: foo=bar; buzz=fizz`. Note that I used a semi-colon, which is the correct delimiter. Here is the RFC spec:

https://tools.ietf.org/html/rfc7540#section-8.1.2.5

```
8.1.2.5.  Compressing the Cookie Header Field

   The Cookie header field [COOKIE] uses a semi-colon (";") to delimit
   cookie-pairs (or "crumbs").  This header field doesn't follow the
   list construction rules in HTTP (see [RFC7230], Section 3.2.2), which
   prevents cookie-pairs from being separated into different name-value
   pairs.  This can significantly reduce compression efficiency as
   individual cookie-pairs are updated.

   To allow for better compression efficiency, the Cookie header field
   MAY be split into separate header fields, each with one or more
   cookie-pairs.  If there are multiple Cookie header fields after
   decompression, these MUST be concatenated into a single octet string
   using the two-octet delimiter of 0x3B, 0x20 (the ASCII string "; ")
   before being passed into a non-HTTP/2 context, such as an HTTP/1.1
   connection, or a generic HTTP server application.

   Therefore, the following two lists of Cookie header fields are
   semantically equivalent.

     cookie: a=b; c=d; e=f

     cookie: a=b
     cookie: c=d
     cookie: e=f
```